src/repr/src/row/encoding.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A permanent storage encoding for rows.
+//!
+//! To minimize cycles spent encoding/decoding, we use Row's internal storage
+//! format, with a prefix to allow for migrations if the internal format
+//! changes.
+//!
+//! ```none
+//! <version: 0u8><v0_encoding: &[u8]>
+//! - Currently the only value for `version` is `0u8`.
+//!
+//! <v0_encoding> is <len_bytes: u64><row_data: &[u8]>
```

Comment:
My instinct is for self-length-ing formats, but in this case, I think it's better to omit it and let whatever framing encoding we use do this. wdyt?

Reply:
I still lean toward doing this, but left it as a TODO. We've got plenty of time to figure this out and I'm having a hard time hashing out the details without yet having taken a swing at a production format for batches of these.